lib/string_decoder.js

```diff
@@ -2,224 +2,203 @@
 
 const Buffer = require('buffer').Buffer;
 
-function assertEncoding(encoding) {
-  // Do not cache `Buffer.isEncoding`, some modules monkey-patch it to support
-  // additional encodings
-  if (encoding && !Buffer.isEncoding(encoding)) {
-    throw new Error('Unknown encoding: ' + encoding);
+// Do not cache `Buffer.isEncoding` when checking encoding names as some
+// modules monkey-patch it to support additional encodings
+function normalizeEncoding(enc) {
```

Comment:
fwiw, I had been experimenting with something similar in the speculative icu module work. See https://github.com/jasnell/node/blob/icu-module/lib/icu.js#L113-L134.